torch/onnx/_internal/fx/passes/readability.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import Dict, List, Sequence, Tuple, Union
+
+import torch
+from torch.onnx._internal import _beartype
+from torch.onnx._internal.fx import _pass, diagnostics
+
+
+class RestoreParameterAndBufferNames(_pass.Transform):
+    """Restore parameter and buffer names from original module.
+
+    This pass is useful for readability of the exported ONNX graph. It restores the
+    parameter and buffer names from the original module. For example, if the original
+    module has a parameter named `root.linear.0.weight`, and the parameter is renamed to
+    `_param_constant9` by FX, this pass will rename it back.
+    """
+    def __init__(
+        self,
+        diagnostic_context: diagnostics.DiagnosticContext,
+        module: torch.fx.GraphModule,
+        original_module: torch.nn.Module,
+    ):
+        super().__init__(diagnostic_context, module)
+        self.original_module = original_module
+
+    @_beartype.beartype
+    def _rename_param_and_buffer(
+        self,
+        diagnostic: diagnostics.Diagnostic,
+        nodes: Sequence[torch.fx.Node],
+        new_name: str,
+    ) -> None:
+        """Rename the parameter/buffer and replace corresponding nodes with new nodes of updated target.
+        """
+        assert len(nodes) > 0, "`nodes` cannot be empty"
+        assert (
+            len({node.target for node in nodes}) == 1
+        ), "`nodes` must all have same `target`"
+        old_name = nodes[0].target
+        assert isinstance(old_name, str), f"Expected str, got type({old_name})"
+        # Parameter/buffer name cannot contain "."
```

Comment:
Sure.

Reply:
Actually I cannot do this in this PR. A bug is revealed that we are naming ONNX initializer by node.name instead of node.target. It has two issues

1. Duplicated initializers if multiple `getattr` nodes were called to retrieve same parameter.
2. `node.name` is not always the same as `node.target`. Previously this is left unnoticed. Uncovered here due to that '/' is not allowed in `node.name`, but allowed in `node.target`. This means we still get `_` in the onnx model initializer names in the end, even if we set to `/` here.

My plan is to merge this PR with `_` to stay consistent, and follow up another PR to fix initializer name, and revert to `/` there.

Reply:
I am actually ok with _ because there will always be a single scope, provided this runs after decomposition. Maybe a double _ instead of a single, if we really want to distinguish from "regular" _ on existing names

Reply:
`_` is a common token in variables. Since the goal is to perserve as much as the original model structure information, `/` is recommanded.